### ISSUE TITLE
document postEditor.setRange and change edit methods to use it

### ISSUE THREAD

**Issue by bantic** [bustlelabs/mobiledoc-kit#258]
#248 added `postEditor.setRange` (and `editor.selectRange`). More of the edit methods in postEditor should be updated to use `setRange` where applicable, and write documentation to describe how to use this API

**Comment by mixonic** [comment]
Things that need to be figured out:

* [ ] `editor.cursor.clearSelection()` How to migrate off
* [ ] `editor.cursor` should have a deprecation message added. Cursor should become private API and no longer have `offsets` on it?
* [ ] `editor.cursor.hasCursor()` How to migrate off

**Comment by rlivsey** [comment]
Some way of getting the current cursor position & setting it again later would be useful if cursor itself is to become private.

I'm currently using the following functions to stash & restore a cursor position when a document is updated externally:

```javascript
function stashCursor(editor) {
  if (!editor || !editor.cursor.hasCursor()) {
    return;
  }

  const { offset, section } = editor.cursor.offsets.head;
  const index = editor.post.sections.toArray().indexOf(section);
  return { index, offset };
}

function reapplyCursor(editor, selection) {
  if (!editor || !selection) {
    return;
  }

  const { index, offset } = selection;

  const section = editor.post.sections.objectAt(index);
  if (!section) {
    return;
  }

  const position = new Position(section, offset);
  const range    = new Range(position, position);

  editor.cursor.selectRange(range);

  editor._reportSelectionState(); // TODO - should we have to do this?
}
```

**Comment by mixonic** [comment]
@rlivsey so what we're doing is moving away from `editor.cursor.offsets` to a new API, `editor.range`.

`editor.cursor.offsets` always reads the cursor position from actual DOM. Once you've done something to modify the abstract post, that position may no longer be coupled to any existing section.

`editor.range`, as you set it, memoizes the range. So after setting, if you ask you'll get back the stored range. This allows us to change the range (and query it) several times during postEditing and flush the position once after rendering.

Some APIs:

* [`Range`](https://github.com/bustlelabs/mobiledoc-kit/blob/master/src/js/utils/cursor/range.js), which is a head and tail position
* [`Position`](https://github.com/bustlelabs/mobiledoc-kit/blob/master/src/js/utils/cursor/position.js)
* [setting and getting range](https://github.com/bustlelabs/mobiledoc-kit/blob/master/src/js/editor/editor.js#L312-L318)
* [`renderRange`](https://github.com/bustlelabs/mobiledoc-kit/blob/master/src/js/editor/editor.js#L286-L303) is called to flush and render the range, for example [after reparsing](https://github.com/bustlelabs/mobiledoc-kit/blob/master/src/js/editor/editor.js#L348)

So we're definitely going to keep a cursor range, but it should settle around `editor.range` and not the `cursor` API. Still questions to answer though.

**Comment by bantic** [comment]
released in [v0.10.4](https://github.com/bustlelabs/mobiledoc-kit/releases/tag/v0.10.4)
